### PR TITLE
[Feature] Add config optn. to disable jail free tp

### LINKED ISSUE #1947: Jail Respawn?
Hey there. 
I have a question regarding the jails. 
Is there a way to remove the auto-tp'ing of players AFTER being jailed?

Say, I do /jail Imdral_Raenwood rochdailcell1 30

Then the player is jailed there, for 30 seconds. But, after, they are tp'd back to their previous location.
I'd like to be able to LEAVE them there. 

So, they're "jailed," for 30 seconds, then, after the time is up, they're unjailed (jail toggles off), but they are left in the cell. 

The key difference is that this allows them to be released/escape, and then they are inconvenienced with having to travel back to their old location. 

How can I turn off the auto-tp?

### PR DESCRIPTION
**Pull request by Ichbinjoe**
Adds the teleport-back-when-freed-from-jail configuration option. When
set to true (default), as with previous versions will teleport the
player which was jailed back to their previous position when freed. When
false, Essentials will not teleport the player anywhere, leaving them
where they are.
    
Closes #1947

### CODE REVIEW

**Comment by mdcfe** [comment]
Presumably when a player is jailed, they are in a closed, confined space. Out of interest, why would you want to leave them in that confined space rather than teleport them out?

File: Essentials/src/com/earth2me/essentials/Settings.java [hunk 1]
```diff
@@ -1439,4 +1439,9 @@ public boolean isDirectHatAllowed() {
     public boolean isConfirmCommandEnabledByDefault(String commandName) {
         return getDefaultEnabledConfirmCommands().contains(commandName.toLowerCase());
     }
+
+    @Override
+    public boolean isTeleportBackWhenFreedFromJail() {
+        return config.getBoolean("teleport-back-when-freed-from-jail", true);
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/Settings.java]: Could you alter this to be consistent with most pre-existing Essentials settings, only reading from the config once and returning the stored value until the config is reloaded?

**Review by mdcfe — CHANGES_REQUESTED** [review]

**Comment by Ichbinjoe** [comment]
> Presumably when a player is jailed, they are in a closed, confined space. Out of interest, why would you want to leave them in that confined space rather than teleport them out?

This is probably a far better question for @ImdralRaenwood (original author of issue). From what it sounded like, there was an interest in making it difficult to travel back to their old location. Maybe what would be better is an overridable release location per-jail?

File: Essentials/src/com/earth2me/essentials/Settings.java (hunk 1 above)
Comment (Ichbinjoe) [reply Essentials/src/com/earth2me/essentials/Settings.java]: Fixed.

**Comment by ImdralRaenwood** [comment]
Exactly that. Immersion.
If I have some player being a dipshit and I jail them, the punishment is good. They can do nothing. But as soon as I ungel them there immediately back at the old location, able to be just as annoying as they were before.

Disabling the return teleport would not only be an inconvenience, forcing them to walk all the way back to the original destination they were at when jailed, but it would maintain immersion. 

I have a medieval fantasy server, and if people are just teleporting back and forth out of jails, it seems a little bit whimsical. Allowing me to disable the return teleport would make the only real teleportation effect the original jailing. 
This way when they are released, just like real people released from real jails, they are still at that jail. They have to leave. Go elsewhere.

I feel it would be a simple introduction to add a true-false to return teleports. 

I would not be opposed to introducing jail-exit tp locations, but, that would not allow for immersion. 

With simply disabling the return teleport, it'd literally mean they have to be released. 
That they have to be unjailed, yes, but also that some must actually flip the lever for the cell's iron doors, allowing them to leave. 

I have the whole jail as a forced survival location, so jailed players have to climb up out of the cavernous Rochdail Keep on survival after a jail, without access to any commands until leaving. 

It increases the punishment for jailed players, and adds detailed immersion.

**Comment by ImdralRaenwood** [comment]
Thank you for this "fixed" message. I am unsure as to it's meaning. What
must I do?

On Apr 2, 2018 10:28 PM, "Joseph Hirschfeld" <notifications@github.com>
wrote:

*@Ichbinjoe* commented on this pull request.
------------------------------

In Essentials/src/com/earth2me/essentials/Settings.java
<https://github.com/EssentialsX/Essentials/pull/1949#discussion_r178716944>:

> @@ -1439,4 +1439,9 @@ public boolean isDirectHatAllowed() {
     public boolean isConfirmCommandEnabledByDefault(String commandName) {
         return
getDefaultEnabledConfirmCommands().contains(commandName.toLowerCase());
     }
+
+    @Override
+    public boolean isTeleportBackWhenFreedFromJail() {
+        return config.getBoolean("teleport-back-when-freed-from-jail", true);

Fixed.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/EssentialsX/Essentials/pull/1949#discussion_r178716944>,
or mute the thread

**Review by mdcfe — APPROVED** [review]

**Comment by mdcfe** [comment]
Once Travis has built this, I'll merge it into the 1.13 branch.

**Comment by mdcfe** [comment]
Closed by mistake due to deleting the 1.13 branch.

**Pull request merged**